### PR TITLE
Protocol updates for Notification Panel Messages in MacOS via APNs

### PR DESCRIPTION
**Pull request by stalkermn**
Hi! I'm looking around on this driver, but i can't find any protocol description for sending APNs to Mac OS Notification panel (Safari based) So, i have some implementation for this. Can u review it.

Here is an example of data which must be sent to APNs for Safari Notifications:

```
{
    "aps": {
        "alert": {
            "title": "Flight A998 Now Boarding",
            "body": "Boarding has begun for Flight A998.",
            "action": "View"
        },
        "url-args": ["boarding", "A998"]
    }
}
```

For more details, please look into
https://developer.apple.com/library/mac/documentation/NetworkingInternet/Conceptual/NotificationProgrammingGuideForWebsites/PushNotifications/PushNotifications.html

### CODE REVIEW

File: src/apns_connection.erl [hunk 1]
```diff
@@ -326,6 +327,21 @@ do_build_payload([{Key, Value} | Params], Payload) ->
...
+                 }->
+
+        AlertJson = {[
+            {<<"title">>, unicode:characters_to_binary(Title)},
+            {<<"body">>, unicode:characters_to_binary(Body)},
+            {<<"action">>, unicode:characters_to_binary(Action)}
+        ]},
+
+        UArgsP = lists:map(fun(Arg) -> unicode:characters_to_binary(Arg) end, UArgs),
+        do_build_payload(Params, [{atom_to_binary(Key, utf8), AlertJson}, {<<"url-args">>, UArgsP} | Payload]);
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 343 is too long:         do_build_payload(Params, [{atom_to_binary(Key, utf8), AlertJson}, {<<"url-args">>, UArgsP} | Payload]);.

File: src/apns_connection.erl [hunk 2]
```diff
@@ -326,6 +327,21 @@ do_build_payload([{Key, Value} | Params], Payload) ->
...
+                  url_args = UArgs
+                 }->
+
+        AlertJson = {[
+            {<<"title">>, unicode:characters_to_binary(Title)},
+            {<<"body">>, unicode:characters_to_binary(Body)},
+            {<<"action">>, unicode:characters_to_binary(Action)}
+        ]},
+
+        UArgsP = lists:map(fun(Arg) -> unicode:characters_to_binary(Arg) end, UArgs),
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 342 is too long:         UArgsP = lists:map(fun(Arg) -> unicode:characters_to_binary(Arg) end, UArgs),.

**Comment by stalkermn** [comment]
Thanks, will fix soon and

**Comment by elbrujohalcon** [comment]
@stalkermn would you mind fixing elvis warnings so that we can merge this pr? Thanks

**Pull request closed without merging**